src/app/patient-dashboard/common/patient-info/edit-address.component.ts

```diff
@@ -1,98 +1,140 @@
   ) {}
-  public ngOnInit(): void {
+
+  public ngOnInit() {
     this.getPatient();
+    this.getLocations();
   }
-  public ngOnDestroy(): void {
-    if (this.subscription.length) {
-      this.subscription.map((sub) => sub.unsubscribe);
+
+  public ngOnDestroy() {
+    if (this.subscriptions.length) {
+      this.subscriptions.map((sub) => sub.unsubscribe);
     }
   }
 
   public getPatient() {
-    const getLocationSubscription = this.locationService
-      .getAmpathLocations()
-      .subscribe((data) => {
-        this.locations = data;
-        console.log(data);
-      });
-    this.subscription.push(getLocationSubscription);
     const getPatientSubscription = this.patientService.currentlyLoadedPatient.subscribe(
       (patient) => {
-        this.patients = new Patient({});
+        this.patient = new Patient({});
         if (patient) {
-          this.patients = patient;
-          if (this.patients.person.preferredAddress) {
-            this.address1 = (this.patients.person
+          this.patient = patient;
+          if (this.patient.person.preferredAddress) {
+            this.address1 = (this.patient.person
               .preferredAddress as any).address1;
-            this.address2 = (this.patients.person
+            this.address2 = (this.patient.person
               .preferredAddress as any).address2;
-            this.address3 = (this.patients.person
+            this.address3 = (this.patient.person
               .preferredAddress as any).address3;
-            this.address7 = (this.patients.person
+            this.address7 = (this.patient.person
               .preferredAddress as any).address7;
-            this.cityVillage = (this.patients.person
+            this.cityVillage = (this.patient.person
               .preferredAddress as any).cityVillage;
-            this.latitude = (this.patients.person
+            this.latitude = (this.patient.person
               .preferredAddress as any).latitude;
-            this.longitude = (this.patients.person
+            this.longitude = (this.patient.person
               .preferredAddress as any).longitude;
-            this.preferredAddressUuid = (this.patients.person
+            this.preferredAddressUuid = (this.patient.person
               .preferredAddress as any).uuid;
           }
         }
       }
     );
-    this.subscription.push(getPatientSubscription);
+    this.subscriptions.push(getPatientSubscription);
+  }
+
+  private getLocations() {
+    const locationResourceServiceSub = this.locationResourceService
+      .getLocations()
+      .pipe(take(1))
+      .subscribe(
+        (locations: any[]) => {
+          const counties: string[] = [];
+          // tslint:disable-next-line:prefer-for-of
+          for (let i = 0; i < locations.length; i++) {
+            this.locations.push({
+              label: locations[i].name,
+              value: locations[i].uuid
+            });
+            counties.push(locations[i].stateProvince);
+          }
```

Comment:
Just a suggestion, you don't have do it. Instead of disabling tslint
```javascript
for (const [i, location] of locations.entries()) {
 this.locations.push({
   label: location.name,
   value: location.uuid
 });
 counties.push(location.stateProvince);
}
```
Of course, you have to enable `downlevelIteration=true` in `tsConfig.json`

Reply:
Good spot, fixed both here and in `PatientCreationComponent`.